drivers/adc/adc_context.h

```diff
@@ -115,6 +115,7 @@ static inline void adc_context_request_next_sampling(struct adc_context *ctx)
 }
 
 #ifdef ADC_CONTEXT_USES_KERNEL_TIMER
+#ifndef ADC_DRIVER_USES_INTERNAL_TIMER
```

Comment:
~This config is not defined~, and is ~also~ not useful from what I see, the APIs here are static inline, they don't take up resources if not used.